docs/modules/ROOT/pages/async-context.adoc

```diff
@@ -6,71 +6,103 @@ ifndef::env-github[]
 and not be required.
 
-NOTE: This maybe seen as a negative in some respects. On the contrary, this does not restrict users from explicitly
-passing state through their APIs.
+To overcome this limitation, we provide two abstractions to propagate context asynchronously: `AsyncContext`, which is
+intended to replace custom `ThreadLocal` definitions, and `CaptureContext` which is used to capture third-party
+context information. This allows programs to decouple values from the exact API call chain by using static APIs to get
+and set values.
 
 However, these benefits don't come for free. Some costs of the static API are as follows:
 
 **Complexity of implementation**
 
+ServiceTalk is a fully asynchronous framework and therefore multiple requests may be multiplexed on the same thread.
+Also, depending on the application's threading model, a single request may be processed on different threads. This means
+that libraries that rely upon ThreadLocal storage such as https://www.slf4j.org/manual.html#mdc[`MDC`] would not work as
+expected.
+
 ServiceTalk is on the hook for making sure the state is saved/restored across asynchronous boundaries. This means
 that ReactiveStreams operators, Executors, and the offloading strategy must account for when relying upon
-`AsyncContext`. In order to accomplish this ServiceTalk's asynchronous control flow needs to account for `AsyncContext`
-and provide tools for users if they have existing code outside ServiceTalk control flow.
+asynchronous context. In order to accomplish this ServiceTalk's asynchronous control flow needs to provide tools for
+users if they have existing code outside ServiceTalk control flow.
 
 **Understandability**
 
 Although asynchronous function composition is not required to use ServiceTalk, the internals of ServiceTalk uses
 asynchronous function composition as a core abstraction which can be difficult to reason about relative to
 traditional imperative programming. On top of this, ServiceTalk provides isolation from the EventLoop
 threads by offloading to application level threads, which introduces asynchronous boundaries. For request/response
-protocols the goal is for the `AsyncContext` to be isolated for each request/response, but folks that need to directly
-interact with `AsyncContext` need to understand how modifications are visible (or not) relative to where they are set.
+protocols the goal is for the asynchronous context to be isolated for each request/response, but folks that need to
+directly interact with asynchronous context need to understand how modifications are visible (or not) relative to where
+they are set or read in their application.
+
+== `AsyncContext`
+
+Although `AsyncContext` may be convenient to use it shouldn't be overused when traditional argument passing is an
+option. The intended use case of `AsyncContext` is to propagate context across method boundaries which do not
```

Comment:
also maybe clarify between method and filter boundaries (just clarifying since I assume a big use case is across filters)?

Reply:
I changed it to API boundaries.

I'm not sure if filters are generally a good use case since our request and response methods carry a `ContextMap` themselves. Within the filter chain, this would be my preferred way to store the data since it's very tightly connected to the request and response objects it will be associated with.